### DISCUSSION TITLE
Use a shared CSS variable for border colors

### DISCUSSION THREAD

**Discussion by agex-n7** [mantinedev/mantine#6448] [Feedback]
While customising the design, I noticed that components reimplement the border style; in particular, they redefine which colours to use for every different component (like the card).
So my suggestion would be to introduce a common CSS variable for this, e.g. `--mantine-color-border`, and use it similarly to `--mantine-color-text`.

I realise that `--mantine-color-default-border` exists, but from what I can tell,  it is only applied to default variants.

If there's interest in this change, then I can introduce the PR for this.

**Comment by rtivital** [comment]
I'd rather not do that for now. It will most likely not work in all cases as border color may change depending on component.

**Comment by andrew-samson** [comment]
@rtivital Can you please tell me if there are any plans to unify the border color in the near future?
Our team is also faced with a variety of gray colors, in particular border colors in components.

At the moment the situation is like this (component: light/dark):
Popover: gray-2/dark-4

*Input: gray-4/dark-4
Checkbox: gray-4/dark-4
Radio: gray-4/dark-4
Pagination: gray-4/dark-4
Button (default): gray-4/dark-4
Alert (default): gray-4/dark-4
Badge (default): gray-4/dark-4

Accordion: gray-3/dark-4
Tabs: gray-3/dark-4
Divider: gray-3/dark-4
Table: gray-3/dark-4
Fieldset: gray-3/dark-4
Notification (With border): gray-3/dark-4
Card (With border): gray-3/dark-4
Kbd: gray-3/dark-4

Different colors in components violate the rule of consistency in design. Also, when developing new components, it is not obvious which shade should be chosen to maintain a unified design system.

Do you have strategies for unification? If yes, we can also offer our PR.
If not, maybe there are design principles regarding color selection?

![image](https://github.com/user-attachments/assets/cdd196c9-6ffc-4fb5-8853-1b4d284f145c)
